### DISCUSSION TITLE
Effects list feature requests

### DISCUSSION THREAD

**Discussion by AlexandreSi** [4ian/GDevelop#3921] [Enhancement ideas (and their technical discussions)]
To improve the effects editor (available for objects and layers), here are a few ideas:
- Add possibility to fold and unfold event content (medium difficulty)
  - When opening an Effects editor, fold all the effects
  - but I don't think it's necessary to store the state as for the events
- Add possibility to copy and paste effects thanks to a toolbar (medium)
- Add a preview of the resource with the effects applied (difficult)

**Comment by Silver-Streak** [comment]
> * Add possibility to fold and unfold event content (medium difficulty)
>   
>   * When opening an Effects editor, fold all the effects
>   * but I don't think it's necessary to store the state as for the events

This is probably important as it brings it in line with how behaviors work now for objects.

> * Add a preview of the resource with the effects applied (difficult)

Would it be harder/easier to have the "preview" just show it on the resource (or layer) in the scene editor?

**Reply by AlexandreSi** [reply]
I think the challenge is in applying the pixi filters to an image preview, adding one on the effects editor should not (and will not) be that hard. And having them on the same screen allows to see the difference when tweaking parameters.

**Reply by 4ian** [reply]
Note that this implies to duplicate all effects so that they run in the editor.
I would rather explore an approach where we run a preview (i.e: the real, game engine) with an auto-generated scene containing just the object, and use the "live preview" feature so that any change in the effects editor is automatically applied to this preview.

**Reply by Silver-Streak** [reply]
I think that makes sense for objects, but doesn't that still leave out layer effects?

**Reply by AlexandreSi** [reply]
> Note that this implies to duplicate all effects so that they run in the editor. I would rather explore an approach where we run a preview (i.e: the real, game engine) with an auto-generated scene containing just the object, and use the "live preview" feature so that any change in the effects editor is automatically applied to this preview.

I thought that `Extensions/Effects/pixi-filters/filter-ascii.js` could be usable, but it I guess it would need to duplicate `Extensions/Effects/ascii-pixi-filter.ts` though.

Having this live preview running besides would make me test all of them. Currently, I have no idea what they do because it's too complicated to try them.
But yeah, it would leave out layer effects.
